core/src/main/java/org/elasticsearch/search/aggregations/support/SingleValuesSourceParser.java

```diff
@@ -0,0 +1,126 @@
+            super(scriptable, formattable, false, ValuesSourceType.BYTES, ValueType.STRING);
+        }
+    }
+
+    public abstract static class GeoPointValuesSourceParser extends SingleValuesSourceParser<ValuesSource.GeoPoint> {
+        protected GeoPointValuesSourceParser(boolean scriptable, boolean formattable) {
+            super(scriptable, formattable, false, ValuesSourceType.GEOPOINT, ValueType.GEOPOINT);
+        }
+    }
+
+    private SingleValuesSourceParser(boolean scriptable, boolean formattable, boolean timezoneAware,
+            ValuesSourceType valuesSourceType, ValueType targetValueType) {
+        super(scriptable, formattable, timezoneAware, valuesSourceType, targetValueType);
+    }
+
+    @Override
+    protected void parseField(XContentParser parser) throws IOException {
+        field = parser.text();
+    }
+
+    @Override
+    protected void parseScript(final String aggregationName, final String currentFieldName,
+            XContentParser parser, ParseFieldMatcher parseFieldMatcher) throws IOException {
+        script = Script.parse(parser, parseFieldMatcher);
+    }
+
+    /**
+     * Creates a {@link SingleValuesSourceAggregatorBuilder} from the information
+     * gathered by the subclass. Options parsed in
+     * {@link SingleValuesSourceParser} itself will be added to the factory
+     * after it has been returned by this method.
+     *
+     * @param aggregationName
+     *            the name of the aggregation
+     * @param valuesSourceType
+     *            the type of the {@link ValuesSource}
+     * @param targetValueType
+     *            the target type of the final value output by the aggregation
+     * @param otherOptions
+     *            a {@link Map} containing the extra options parsed by the
+     *            {@link #token(String, String, org.elasticsearch.common.xcontent.XContentParser.Token,
+     *             XContentParser, ParseFieldMatcher, Map)}
+     *            method
+     * @return the created factory
+     */
+    protected abstract SingleValuesSourceAggregatorBuilder<VS, ?> createFactory(String aggregationName, ValuesSourceType valuesSourceType,
+            ValueType targetValueType, Map<ParseField, Object> otherOptions);
+
+    @Override
+    protected SingleValuesSourceAggregatorBuilder<VS, ?> createBuilder(String aggregationName, final ValueType parsedValueType,
+            final String parsedFormat, final Object parsedMissing, final DateTimeZone parsedTimeZone,
+            Map<ParseField, Object> otherOptions) {
+        SingleValuesSourceAggregatorBuilder<VS, ?> factory = createFactory(aggregationName, this.valuesSourceType, this.targetValueType,
+            otherOptions);
+        if (field != null) {
+            factory.field(field);
+        }
+        if (script != null) {
+            factory.script(script);
+        }
+        if (parsedValueType != null) {
+            factory.valueType(parsedValueType);
+        }
```

Comment:
Can we not set this and the following attributes in the superclass after this method has returned?